src/translations/en.json

```
@@ -2641,6 +2641,7 @@
           },
           "delete": "Delete",
           "confirm_delete": "Are you sure you want to delete this device?",
+          "confirm_delete_integration": "Are you sure you want to remove {integration} from this device?",
```

Isn't it the other way around from a user perspective?

```suggestion
          "confirm_delete_integration": "Are you sure you want to remove this device from {integration}?",
```

---

I don't know, I reused the language from the button label which is:
```
"delete_device_integration": "Remove {integration} from device"
```

---

Since you're doing the removal from the device card your suggestion might make the user think the entire device will be removed. Initial suggestion explains it better to the user

---

I agree with Frenck.

Removing a device means we are telling MQTT to remove this device (`async_remove_config_entry_device` callback in core). 

If you aligned it with the button, let's update the button label too 👍

---

Alright, updated